iotedgedev/deploymentmanifest.py

```diff
@@ -147,3 +186,74 @@ def _get_module_content(self):
+            self.output.info("Errors found in deployment manifest schema, skip schema validation. Error:%s" % schemaErr.message)
+        except Exception as ex:  # Ignore other non schema validation errors
+            self.output.info("Unexpected error during deployment manifest schema validation, skip schema validation. Error:%s" % ex)
+
+        return validation_success
+
+    # Call _validate_deployment_manifest_schema first. This function assumes createOptions are strings.
+    def _validate_createOptions(self):
+        self.output.info("Start validating createOptions for all modules.")
+        modules = self.get_all_modules()
+        validation_success = True
+        for module_name, module_info in modules.items():
+            try:
+                self.output.info("Validating createOptions for module %s" % module_name)
+                create_options = []
+                if "settings" in module_info and "createOptions" in module_info["settings"]:
+                    create_options_value = module_info["settings"]["createOptions"]
+                    if len(create_options_value) > TWIN_VALUE_MAX_SIZE:
+                        validation_success = False
+                        self.output.info("Length of createOptions in module %s exceeds %d" % (module_name, TWIN_VALUE_MAX_SIZE))
+                    create_options.append(create_options_value)
+                    # Merge additional create options
+                    for i in range(1, TWIN_VALUE_MAX_CHUNKS):
+                        property_name = "createOptions0%d" % i
+                        if property_name in module_info["settings"]:
+                            create_options_value = module_info["settings"][property_name]
+                            if len(create_options_value) > TWIN_VALUE_MAX_SIZE:
+                                validation_success = False
+                                self.output.info("Length of %s in module %s exceeds %d" % (property_name, module_name, TWIN_VALUE_MAX_SIZE))
+                            create_options.append(create_options_value)
+                        else:
+                            break
+                    # Verify createOptions is valid json
+                    create_options_string = "".join(create_options).strip()
+                    if not create_options_string.startswith('{'):
+                        validation_success = False
+                        self.output.info("createOptions of module %s should be an object" % module_name)
+                    else:
+                        try:
+                            json.loads(create_options_string)
+                            self.output.info("createOptions of module %s validation passed" % module_name)
+                        except ValueError as err:
+                            validation_success = False
+                            self.output.info("createOptions of module %s is not a valid JSON string. Error: %s" % (module_name, err))
```

Comment:
can we output warning instead of info?

Reply:
Actually there're only 2 output channels: stdout and stderr. Currently I outputs the validation failures to stdout channel because Azure Pipelines force print messages in stderr in build summary and there's option to fail the task if any messages arrive stderr channel.

Reply:
"Warning" does not mean the output channel. Instead, we should have a function in output class to specify a kind of warning message. Like [Warning] prefix